components/nimbus/src/lib.rs

```diff
@@ -100,13 +102,22 @@ impl NimbusClient {
             available_randomization_units,
             targeting_attributes: app_context.clone().into(),
         });
+
+        let db: OnceCell<Database> = OnceCell::default();
+
+        let db_path = db_path.into();
+
+        let event_store = Mutex::new(EventStore::from_persisted_data(
```

Comment:
It could probably be evaluated in a more lazy manner, let me see if I can emulate here to be more like what happens with the experiments and enrollments.